### PR TITLE
Document different regorules for v414 and v428 due to authorizer changes

### PR DESCRIPTION
**Pull request by maltesander**
# Description

Not sure we want to point to pull requests and docs referenced in there...any opinions?

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [ ] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: docs/modules/trino/pages/usage-guide/security.adoc [hunk 1]
```diff
@@ -74,9 +74,20 @@ In the Trino CLI and web interface, LDAP users can now be used to log in.
 In order to authorize Trino via OPA, a ConfigMap containing a rego rule package for Trino has to be applied and referenced in the TrinoCluster resource.
 The following example is an all-access Rego rule for testing with the user `admin`. Do not use it in production!
 
+NOTE: Due to changes in the Trino OPA authorizer, rego rules differ from Version 414 and 428. The OPA authorizer will become part of the Trino package once this https://github.com/trinodb/trino/pull/19532[pull request] is merged. The stackable Trino version 428 already uses that new OPA authorizer and utilizes https://github.com/bloomberg/trino/blob/add-open-policy-agent/plugin/trino-opa/README.md#batch-mode[batch mode]. This is under active development and is not stable yet!
```
Comment (sbernauer) [inline docs/modules/trino/pages/usage-guide/security.adoc:77]:
```suggestion
NOTE: Due to changes in the Trino OPA authorizer, rego rules differ between version 414 and 428. The OPA authorizer will become part of the Trino package once this https://github.com/trinodb/trino/pull/19532[pull request] is merged. The stackable Trino version 428 already uses that new OPA authorizer and utilizes https://github.com/bloomberg/trino/blob/add-open-policy-agent/plugin/trino-opa/README.md#batch-mode[batch mode]. This is under active development and is not stable yet!
```

**Review by sbernauer — COMMENTED** [review]
I'm fine with linking a PR, but I would use a permalink for the docs when possible

**Review by sbernauer — APPROVED** [review]
LGTM, thx!

**Comment by lfrancke** [comment]
This is a temporary thing, right? Do we have/need a tracking issue to remove this again?

**Comment by maltesander** [comment]
> This is a temporary thing, right? Do we have/need a tracking issue to remove this again?

Yeah the 414 part (our old authorizer) can be removed once we do not support that anymore. Where do you want to track it?

**Comment by lfrancke** [comment]
I think a follow-up issue linked to this one would be good.
I can then label it with scheduled-for/next (doesn't exist yet but I can take care of that)

**Comment by maltesander** [comment]
https://github.com/stackabletech/trino-operator/issues/496

**Pull request merged**